src/ess/reflectometry/conversions.py

```diff
@@ -126,126 +117,39 @@ def specular_reflection(
-    return WavelengthData[Run](data_array_wav)
-
-
-def wavelength_to_theta(
-    data_array: WavelengthData[Run],
+def add_coords(
+    da: ChopperCorrectedTofEvents[Run],
     graph: SpecularReflectionCoordTransformGraph[Run],
-) -> ThetaData[Run]:
-    """
-    Use :code:`transform_coords` to find the theta values for the events and
-    potentially add ORSO metadata.
-
-    Parameters
-    ----------
-    data_array:
-        Data array to convert.
-    graph:
-        Graph for :code:`transform_coords`.
-
-    Returns
-    -------
-    :
-        New data array with theta coordinate.
-    """
-    data_array_theta = data_array.transform_coords(['theta'], graph=graph)
-    #    # Determine if 'gravity' is in the graph and if to add the gravity correction
-    #    if any(
-    #        [
-    #            'gravity' in i.parameters.keys()
-    #            for i in map(inspect.signature, graph.values())
-    #        ]
-    #    ):
-    #        data_array_theta.attrs['orso'].value.reduction.corrections += [
-    #            'gravity correction'
-    #        ]
-    # except ImportError:
-    #    orso.not_found_warning()
-    return ThetaData[Run](data_array_theta)
-
-
-def theta_to_q(
-    data_array: FootprintCorrectedData[Run],
-    q_bins: QBins,
-    graph: SpecularReflectionCoordTransformGraph[Run],
-) -> QData[Run]:
-    """
-    Convert from theta to Q and if necessary bin in Q.
-
-    Parameters
-    ----------
-    data_array:
-        Data array to convert.
-    q_edges:
-        The lower and upper limits for the Q.
-    graph:
-        Graph for :code:`transform_coords`.
-
-    Returns
-    -------
-    :
-        New data array with theta coordinate.
-    """
-    data_array = data_array.transform_coords(["Q"], graph=graph)
-    data_array = data_array.bin({q_bins.dim: q_bins})
-    return QData[Run](data_array)
-
-
-def histogram(data_array: QData[Run]) -> HistogrammedQData[Run]:
-    """
-    Sum the event bins.
-
-    Parameters
-    ----------
-    data_array:
-        Data array to be summed.
-
-    Returns
-    -------
-    :
-        Summed data array.
-    """
-    return HistogrammedQData[Run](data_array.hist())
+) -> FullData[Run]:
+    da = da.transform_coords(["theta", "wavelength", "Q"], graph=graph)
+    da.coords['z_index'] = da.coords['detector_number'] // sc.scalar(64, unit=None)
+    da.coords['y_index'] = da.coords['detector_number'] % sc.scalar(64, unit=None)
+    return da
+
+
+def add_masks(
+    da: FullData[Run], ylim: YIndexLimits, wb: WBins, zlim: ZIndexLimits
+) -> MaskedFullData[Run]:
+    da.masks['y_index_range'] = (da.coords['y_index'] < ylim[0]) | (
+        da.coords['y_index'] > ylim[1]
+    )
+    da.masks['wavelength_mask'] = (da.coords['wavelength'] < wb[0]) | (
+        da.coords['wavelength'] > wb[-1]
+    )
+    da.masks['z_index_range'] = (da.coords['z_index'] < zlim[0]) | (
+        da.coords['z_index'] > zlim[1]
+    )
+    return da
```

Comment:
I think there are likely a bunch of changes in `ess/reflectometry` which now only work with the way the Amor data is loaded, i.e., without grouping by detector number. Has it been determined that we clearly do *not* want to work with detector-number grouping? Or is this only temporary for the Amor workflow?

Reply:
It's temporary, as mentioned in https://github.com/scipp/essreflectometry/pull/44#discussion_r1592053075

I agree a lot of stuff here is amor specific and should be moved to the amor module.